pimcore/config/helper.php

```diff
@@ -126,53 +126,54 @@ function object2array($node)
 
 /**
  * @param  $args
- * @return bool|string
+ * @return false|string
  */
 function array_urlencode($args)
 {
     if (!is_array($args)) {
         return false;
     }
-    $out = '';
-    foreach ($args as $name => $value) {
-        if (is_array($value)) {
-            foreach ($value as $key => $val) {
-                $out .= urlencode($name).'['.urlencode($key).']'.'=';
-                $out .= urlencode($val).'&';
-            }
-        } else {
-            $out .= urlencode($name).'=';
-            $out .= urlencode($value).'&';
-        }
-    }
-
-    return substr($out, 0, -1); //trim the last & }
+    
+    return str_replace(['%5B', '%5D'], ['[', ']'], http_build_query($args));
```

Comment:
Maybe the best solution is just call the http_build_query function without the str_replace?

Do you know in which cases pimcore needs the returned string of this function or is the method only used for output?

I open the following links in chrome and firefox:
* /?one%5Blevel2%5D%5Bvalue%5D=%5BmyValue%5D
* /?one[level2][value]=[myValue]
and i got the same result for var_dump( $_GET ). I think it does not matter if you call a link with [ and ] or with %5B and %5D.

I would prefer to delete the str_replace call before http_build_query, but if u want i will change it to the array_toquerystringHelper method.

It is your choice just tell me what to do and I will do it :)